mitiq/pt/tests/test_pt.py

```diff
@@ -144,3 +144,20 @@ def test_no_CNOT_CZ_circuit(twirl_func):
 
     for i in range(5):
         assert _equal(circuit, twirled_output[i])
+
+
+@pytest.mark.parametrize(
+    "noise_op", [cirq.bit_flip(p=0.1), cirq.depolarize(p=0.01)]
+)
+def test_noisy_CNOT_CZ_circuit(noise_op):
```

Comment:
Can you check if this works on other Mitiq-supported circuit types?

Reply:
Hi @Misty-W! Well, that was a very good question, because indeed it does not work. the biggest problem here is that the Cirq function `to_qasm ` does not support noise operators. 

And even if this was the case, adding noise on Qiskit is done on execution (AFAIK). 

A similar issue happens on `QuantumTapes` on Pennylane with the `to_openqasm ` method. So also converting from a Pennylane noise channel to Cirq is also not trivial.

I honestly do not know how to proceed there 🤔

Reply:
Hi, @purva-thakre. So you mean have some sort of look up dictionary based on the string and the circuit module? I guess that means trying to include as many noise channels as possible as long as all the supported `QPROGRAM`s have an equivalent, right?

This would still excluded qiskit as noise is added when executing the circuit. Is this correct?

Reply:
> So you mean have some sort of look up dictionary based on the string and the circuit module? I guess that means trying to include as many noise channels as possible as long as all the supported QPROGRAMs have an equivalent

Yes, this is what I was thinking but I don't know if this is the best possible solution. I am sure this will get discussed during the community call tomorrow. 

> This would still excluded qiskit as noise is added when executing the circuit

Did this change with the latest version of Qiskit? As far as I know, Qiskit used to allow a user to selectively add noise to a circuit.

Reply:
I have added some minimal functionalities that work now with Cirq and Pennylane using your suggestion. I still think that it is kind of cumbersome to approach it like this, but since noise is not supported in QASM conversions, I cannot think of a better way.

I added the examples as unittests.

If you think this is the way to go, I can go ahead and register more dispatches for the other supported QPROGRAMS and add more noise models (the idea would be to include the ones supported by Cirq as well and raise an error if it's something else)

Please let me know what you think! 😄

Reply:
Thank you for the update! Any feedback on my proposed solution would also be great. For the moment I will assume there is a better way, so I will wait to hear from you 😊.